src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy

```diff
@@ -209,45 +211,41 @@ abstract class ComposeUp extends DefaultTask {
         }
     }
 
-    private static final VOLATILE_STATE_KEYS = ['RunningFor']
-    private static final UNSTABLE_ARRAY_STATE_KEYS = ['Mounts', 'Ports', 'Networks', 'Labels', 'Publishers']
-
     @Internal
     protected def getStateForCache() {
-        String processesAsString = composeExecutor.get().execute('ps', '--format', 'json')
-        String processesState = processesAsString
+        def state = []
+
         try {
-            // Since Docker Compose 2.21.0, the output is not one JSON array but newline-separated JSONs.
-            Map<String, Object>[] processes
-            if (processesAsString.startsWith('[')) {
-                processes = new JsonSlurper().parseText(processesAsString)
-            } else {
-                processes = processesAsString.split('\\R').findAll { it.trim() }.collect { new JsonSlurper().parseText(it) }
-            }
-            List<Object> transformed = processes.collect {
-                // Status field contains something like "Up 8 seconds", so we have to strip the duration.
-                if (it.containsKey('Status') && it.Status.startsWith('Up ')) it.Status = 'Up'
-                VOLATILE_STATE_KEYS.each { key -> it.remove(key) }
-                UNSTABLE_ARRAY_STATE_KEYS.each { key -> it[key] = parseAndSortStateArray(it[key]) }
-                it
-            }
-            processesState = transformed.join('\t')
-        } catch (Exception e) {
-            logger.warn("Cannot process JSON returned from 'docker compose ps --format json'", e)
-        }
-        processesState + composeExecutor.get().execute('config') + startedServices.get().join(',')
-    }
+            def containersIds = composeExecutor.get()
+                .execute('ps', '--quiet', '--all')
+                .readLines()
+            def containersInfos = new JsonSlurper().parseText(
+                dockerExecutor.execute(['inspect', *containersIds] as String[])
+            )
+            def jsonGenerator = new JsonGenerator.Options()
+                .excludeFieldsByName('Log')
+                .addConverter(List) { items, key ->
+                    if (key in [null, 'Args']) {
+                        return items
+                    }
 
-    protected Object parseAndSortStateArray(Object list) {
-        if (list instanceof List) {
-            //Already provided as a List, no pre-parsing needed
-            return list.sort { (first, second) -> first.toString() <=> second.toString() }
-        } else if (list instanceof String && list.contains(",")) {
-            //Not already a list, but a comma separated string
-            return list.split(',').collect { it.trim() }.sort()
-        } else {
-            return list
+                    return items.sort()
+                }
```

Comment:
Please add a comment for this code block 🙏

Reply:
@augi I have completed the comments for this block. Is it now readable and sufficient?

Reply:
Yes, perfect! 👍